### ISSUE TITLE
reamber.osu.OsuMap.OsuMap.rate not working as expected 

### ISSUE THREAD

**Issue by Bestfast** [Eve-ning/reamberPy#36]
**Describe the bug**

When using reamber.osu.OsuMap.OsuMap.rate the output is a corrupted file. Example:

```
RATE = 1.5
FILE_PATH = "E:\Games\osu!\Songs\992889 4K Chordjack Challenge\Various Artists - 4K Chordjack Challenge (Onta_Bekasi) [BRIGHT - 1 year 2 months 20 days].osu"
m = OsuMap.readFile(FILE_PATH)
m.rate(RATE, True)
m.version = m.version + " {}x".format(RATE)
m.writeFile(file_path[:-4] + " {}.osu".format(RATE))
```
(the audio file has been externally processed with ffmpeg)
output: https://del.dog/tydeamucyg.txt

replacing last 4 lines from:

> 192,192,26464,128,0,35453:0:0:0:0:
> 448,192,26632,128,0,35677:0:0:0:0:
> 192,192,26800,128,0,35901:0:0:0:0:
> 64,192,26968,128,0,36125:0:0:0:0:

to:

> 192,192,26464,1,0,0:0:0:0:
> 448,192,26632,1,0,0:0:0:0:
> 192,192,26800,1,0,0:0:0:0:
> 64,192,26968,1,0,0:0:0:0:

"fixes" it, even though the result won't be an uprated file.

**Files used (if any)**

https://osu.ppy.sh/beatmapsets/992889#mania/2076376 
![screenshot1862](https://user-images.githubusercontent.com/31173611/117548087-3ee5cb80-b033-11eb-830d-4fa3b6019087.jpg)

**Comment by Bestfast** [comment]
After further testing, the same problem happens with Quaver Map objects. Quaver log outputs:


> [1:42:19] - RUNTIME - ERROR: YamlDotNet.Core.YamlException: (Line: 29, Col: 14, Idx: 669) - (Line: 29, Col: 31, Idx: 686): Exception during deserialization
>  ---> System.FormatException: Input string was not in a correct format.
>    at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
>    at System.Number.ParseUInt64(ReadOnlySpan`1 value, NumberStyles styles, NumberFormatInfo info)
>    at System.UInt64.Parse(String s)
>    at YamlDotNet.Serialization.NodeDeserializers.ScalarNodeDeserializer.DeserializeIntegerHelper(TypeCode typeCode, String value)
>    at YamlDotNet.Serialization.NodeDeserializers.ScalarNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
>    at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    --- End of inner exception stack trace ---
>    at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass3_0.<DeserializeValue>b__0(IParser r, Type t)
>    at YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
>    at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass3_0.<DeserializeValue>b__0(IParser r, Type t)
>    at YamlDotNet.Serialization.NodeDeserializers.CollectionNodeDeserializer.DeserializeHelper(Type tItem, IParser parser, Func`3 nestedObjectDeserializer, IList result, Boolean canUpdate)
>    at YamlDotNet.Serialization.NodeDeserializers.CollectionNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
>    at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass3_0.<DeserializeValue>b__0(IParser r, Type t)
>    at YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
>    at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
>    at YamlDotNet.Serialization.Deserializer.Deserialize(IParser parser, Type type)
>    at YamlDotNet.Serialization.Deserializer.Deserialize(TextReader input, Type type)
>    at Quaver.API.Maps.Qua.Parse(String path, Boolean checkValidity)
>    at Quaver.Shared.Database.Maps.MapsetImporter.InsertAndUpdateSelectedMap(String extractDirectory)

I'll test more and post everything here, but the issue seems to be in reamber.base.Map.Map.rate (?)

**Comment by Eve-ning** [comment]
Found 2 Issues:
1) Long Note Length is incorrectly rated
2) If PreviewTime is a float
e.g.: 1000.0 will break it.
e.g.: 1000 will work

**Comment by Eve-ning** [comment]
Fixed on 0.0.19a
